commons-mongo/src/main/scala/com/avsystem/commons/mongo/typed/TypedMongoCollection.scala

```diff
@@ -0,0 +1,292 @@
+package com.avsystem.commons
+package mongo.typed
+
+import com.avsystem.commons.mongo.core.GenCodecRegistry
+import com.mongodb.bulk.BulkWriteResult
+import com.mongodb.client.model._
+import com.mongodb.client.result.{DeleteResult, UpdateResult}
+import com.mongodb.reactivestreams.client.{DistinctPublisher, FindPublisher, MongoCollection}
+import com.mongodb.{MongoNamespace, ReadConcern, ReadPreference, WriteConcern}
+import monix.eval.Task
+import monix.reactive.Observable
+import org.bson.codecs.configuration.CodecRegistry
+import org.bson.{BsonDocument, BsonValue}
+import org.reactivestreams.Publisher
+
+final class TypedMongoCollection[E <: BaseMongoEntity : MongoAdtFormat](
```

Comment:
You wouldn't expect me to say this, but I'd consider making this non-final. There are some cases where extending this with a bit of logic could serve as a simple yet complete persistence API, without further layers of composition.

Reply:
I understand but if you want me to maintain binary compat in a proper way, that might become annoying.

Reply:
That's up to you to decide then, I'm not expecting too much in terms of compatibility when I'm upgrading commons :D